From 889936ac2a4c5483af7dbdeba599832d95e7d299 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Sun, 26 Mar 2023 17:20:49 +0200
Subject: [PATCH 1/4] u-boot-qoriq: Enable VID for LTC7132

This means adding boolean defines for this PMIC and also permitting
to reconfigure the PWM_CHANNEL0 value.

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 board/freescale/common/Kconfig | 12 ++++++++++++
 board/freescale/common/vid.c   |  9 ++++++++-
 board/freescale/common/vid.h   |  2 ++
 3 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/board/freescale/common/Kconfig b/board/freescale/common/Kconfig
index 77d5ca722c..2446d7e738 100644
--- a/board/freescale/common/Kconfig
+++ b/board/freescale/common/Kconfig
@@ -85,6 +85,18 @@ config VOL_MONITOR_LTC3882_SET
 	 This option enables LTC3882 voltage monitor set
 	 functionality. It is used by the common VID driver.
 
+config VOL_MONITOR_LTC7132_READ
+	bool "Enable the LTC7132 voltage monitor read"
+	help
+	 This option enables LTC7132 voltage monitor read
+	 functionality. It is used by the common VID driver.
+
+config VOL_MONITOR_LTC7132_SET
+	bool "Enable the LTC7132 voltage monitor set"
+	help
+	 This option enables LTC7132 voltage monitor set
+	 functionality. It is used by the common VID driver.
+
 config VOL_MONITOR_ISL68233_READ
 	bool "Enable the ISL68233 voltage monitor read"
 	help
diff --git a/board/freescale/common/vid.c b/board/freescale/common/vid.c
index d2c9bbbfe9..5ea259b309 100644
--- a/board/freescale/common/vid.c
+++ b/board/freescale/common/vid.c
@@ -250,8 +250,10 @@ static int read_voltage_from_IR(int i2caddress)
 
 #if defined(CONFIG_VOL_MONITOR_ISL68233_READ) || \
 	defined(CONFIG_VOL_MONITOR_LTC3882_READ) || \
+	defined(CONFIG_VOL_MONITOR_LTC7132_READ) || \
 	defined(CONFIG_VOL_MONITOR_ISL68233_SET) || \
-	defined(CONFIG_VOL_MONITOR_LTC3882_SET)
+	defined(CONFIG_VOL_MONITOR_LTC3882_SET) || \
+	defined(CONFIG_VOL_MONITOR_LTC7132_SET)
 
 /*
  * The message displayed if the VOUT exponent causes a resolution
@@ -302,6 +304,7 @@ static int get_pmbus_multiplier(DEVICE_HANDLE_T dev)
 #endif
 
 #if defined(CONFIG_VOL_MONITOR_ISL68233_READ) || \
+	defined(CONFIG_VOL_MONITOR_LTC7132_READ) || \
 	defined(CONFIG_VOL_MONITOR_LTC3882_READ)
 static int read_voltage_from_pmbus(int i2caddress)
 {
@@ -347,6 +350,7 @@ static int read_voltage(int i2caddress)
 #elif defined CONFIG_VOL_MONITOR_IR36021_READ
 	voltage_read = read_voltage_from_IR(i2caddress);
 #elif defined(CONFIG_VOL_MONITOR_ISL68233_READ) || \
+	  defined(CONFIG_VOL_MONITOR_LTC7132_READ) || \
 	  defined(CONFIG_VOL_MONITOR_LTC3882_READ)
 	voltage_read = read_voltage_from_pmbus(i2caddress);
 #else
@@ -458,6 +462,7 @@ static int set_voltage_to_IR(int i2caddress, int vdd)
 #endif
 
 #if defined(CONFIG_VOL_MONITOR_ISL68233_SET) || \
+	defined(CONFIG_VOL_MONITOR_LTC7132_SET) || \
 	defined(CONFIG_VOL_MONITOR_LTC3882_SET)
 static int set_voltage_to_pmbus(int i2caddress, int vdd)
 {
@@ -527,6 +532,7 @@ static int set_voltage(int i2caddress, int vdd)
 #ifdef CONFIG_VOL_MONITOR_IR36021_SET
 	vdd_last = set_voltage_to_IR(i2caddress, vdd);
 #elif defined(CONFIG_VOL_MONITOR_ISL68233_SET) || \
+	  defined(CONFIG_VOL_MONITOR_LTC7132_SET) || \
 	  defined(CONFIG_VOL_MONITOR_LTC3882_SET)
 	vdd_last = set_voltage_to_pmbus(i2caddress, vdd);
 #else
@@ -675,6 +681,7 @@ int adjust_vdd(ulong vdd_override)
 	debug("VID: Core voltage is currently at %d mV\n", vdd_last);
 
 #if defined(CONFIG_VOL_MONITOR_LTC3882_SET) || \
+	defined(CONFIG_VOL_MONITOR_LTC7132_SET) || \
 	defined(CONFIG_VOL_MONITOR_ISL68233_SET)
 	/* Set the target voltage */
 	vdd_current = set_voltage(i2caddress, vdd_target);
diff --git a/board/freescale/common/vid.h b/board/freescale/common/vid.h
index b34c080b4b..8cdc1bcdca 100644
--- a/board/freescale/common/vid.h
+++ b/board/freescale/common/vid.h
@@ -52,7 +52,9 @@
 #endif
 
 /* PM Bus commands code for LTC3882*/
+#ifndef PWM_CHANNEL0
 #define PWM_CHANNEL0                    0x0
+#endif
 #define PMBUS_CMD_PAGE                  0x0
 #define PMBUS_CMD_READ_VOUT             0x8B
 #define PMBUS_CMD_VOUT_MODE			0x20
-- 
2.34.1

